lib/propcheck.ex

```diff
@@ -429,20 +429,17 @@ defmodule PropCheck do
               unquote(prop)
             rescue
               e in ExUnit.AssertionError ->
-                stacktrace = System.stacktrace
                 if verbose? do
                   e |> ExUnit.AssertionError.message() |> IO.write()
-                  formatted = Exception.format_stacktrace(stacktrace)
+                  formatted = Exception.format_stacktrace(__STACKTRACE__)
```

Comment:
I would suggest a downward compatible solution: Check the version of Elixir at compile time in a macro and put it into here. Something like 

```` elixir
    # outside a function: 
    if Version.match(System.version, ">= 1.7.0") do 
        @STACK __STACKTRACE__
    else 
        @STACK System.stacktrace()
    end
````

This can be used instead of `__STACKTRACE__`.

Reply:
Seems it's not possible to use that macro outside the `try-catch` clause. [(elixir-lang)](https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/kernel/special_forms.ex#L718)

Defining inside the function like it was before:
```
rescue
  e in ExUnit.AssertionError ->
    stacktrace = if Version.match?(System.version, ">= 1.7.0") do
      __STACKTRACE__
    else
      System.stacktrace
    end

    if verbose? do
      e |> ExUnit.AssertionError.message() |> IO.write()
      formatted = Exception.format_stacktrace(stacktrace)
      IO.puts("stacktrace:\n#{formatted}")
    end
...
```

will still raise warnings, e.g.
```
warning: System.stacktrace/0 is deprecated, use __STACKTRACE__ instead
  test/store_counter_examples_test.exs:14
```

Reply:
Yes, I got the same results with my approach. That your approach does not work is clear, since both branches of the `if` are compiled hence the warning - my approach was trying to be something like conditional compiling. That would work, if `__STACKTRACE__` is allowed outside of a try-catch clause. So let us look for a different approach.